email_account_validity/_base.py

```diff
@@ -102,38 +104,45 @@ async def send_renewal_email(self, user_id: str, expiration_ts: int):
             display_name = profile.display_name
             if display_name is None:
                 display_name = user_id
-        except StoreError:
+        except SynapseError:
             display_name = user_id
 
         renewal_token = await self.generate_renewal_token(user_id)
 
-        url = "%s_synapse/client/email_account_validity/renew?token=%s" % (
-            self._api.public_baseurl,
-            renewal_token,
-        )
+        if self._send_links:
+            url = "%s_synapse/client/email_account_validity/renew?token=%s" % (
+                self._api.public_baseurl,
+                renewal_token,
+            )
+        else:
+            # If we're not supposed to send a URL, fallback to the URL being just the
+            # token. Templates should be using the `renewal_token` variable, but we do
+            # this so old templates don't break.
+            url = renewal_token
```

Comment:
maybe better just to set it to the empty string or None or something. the `renewal_token` seems pretty arbitrary.